File: question/neetcode150/trees/invert-binary-tree.py
from typing import Optional

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        stack = [root]

        while stack:
            n = stack.pop()

            if n:
                n.left, n.right = n.right, n.left
                stack.append(n.left)
                stack.append(n.right)

        return root

File: question/neetcode/binary_tree/invert-binary-tree.cpp
#include <bits/stdc++.h>
using namespace std;

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    public:
        void dfs(TreeNode* node){
            if(not node) return;

            swap(node->left, node->right);
            dfs(node->left);
            dfs(node->right);
        }
        TreeNode* invertTree(TreeNode* root) {
            dfs(root);

            return root;
        }
    };